samples/api-doc-search/azure-function-nodejs/src/functions/embedFunctions.js

```diff
@@ -0,0 +1,348 @@
+require('dotenv').config();
+const WordExtractor = require("word-extractor");
+const appInsights = require('applicationinsights');
+
+// Configure Application Insights with your instrumentation key or connection string
+const instrumentationKey = process.env.APPINSIGHTS_INSTRUMENTATIONKEY;
+
+// Connection string
+const connectionString = process.env.APPINSIGHTS_CONNECTIONSTRING;
+
+appInsights.setup(connectionString || instrumentationKey)
+  .setAutoCollectRequests(true)
+  .setAutoCollectPerformance(true, true)
+  .setAutoCollectExceptions(true)
+  .setAutoCollectDependencies(true)
+  .setAutoCollectConsole(true, true)
+  .setUseDiskRetryCaching(true)
+  .start();
+
+const appInsightsClient = appInsights.defaultClient;
+
+// Azure Storage connection string and container name.
+const azureStorageConnString = process.env.AzureStorageDBConnString;
+const azureBlobContainerName = process.env.AzureBlobContainerName;
+
+// Azure Open AI service endpoint, API key, and deployment name.
+const azureAiApiEndpoint = process.env.AzureOpenAIEndpoint;
+const openAIApiKey = process.env.AzureOpenAIApiKey;
+const openAIDeploymentName = process.env.AzureOpenAIDeploymentName; // Ensure this is correct
+
+// Create an instance of the Azure OpenAI client
+const azureOpenAIClient = new OpenAIClient(azureAiApiEndpoint, new AzureKeyCredential(openAIApiKey));
+
+// Cosmos DB configuration.
+const cosmosEndpoint = process.env.CosmosDBEndpoint;
+const cosmosKey = process.env.CosmosDBKey;
+
+const cosmosClientOptions = {}; // Optional client options
+const cosmosClient = new CosmosClient({ endpoint: cosmosEndpoint, key: cosmosKey }, cosmosClientOptions);
+
+// Cosmos DB database and container IDs.
+const databaseId = process.env.CosmosDBDatabaseId;
+const containerId = process.env.CosmosDBContainerId;
+
+// Get a reference to the Cosmos DB container.
+const container = cosmosClient.database(databaseId).container(containerId);
+
+/**
+ * Function to split the text content of a file into smaller chunks.
+ *
+ * @param {string} fileContentsAsString - The content of the file as a single string.
+ * @returns {Array} - An array of objects, each containing a chunk of the original text.
+ * @throws Will throw an error if text splitting fails.
+ */
+async function splitText(fileContentsAsString) {
+  // Create an instance of the RecursiveCharacterTextSplitter with specified options
+  const splitter = new RecursiveCharacterTextSplitter({
+    chunkSize: 800,
+    chunkOverlap: 10,
+    separators: ['\n\n', '\n', ' ', '']
+  });
+
+  // Split the file content into smaller chunks and return the result
+  return await splitter.createDocuments([fileContentsAsString]);
+}
+
+/**
+ * Function to list all blobs in a container and retrieve their URLs.
+ *
+ * @returns {Array} - An array containing URLs of all blobs in the container.
+ * @throws Will throw an error if listing blobs fails.
+ */
+async function InitiateEmbeddings(blobUrl, context) {
```

Comment:
method summary missing Param.

Reply:
Updated method summary for all the methods.